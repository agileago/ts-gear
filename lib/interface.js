"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const types_1 = require("./tool/types");
exports.httpMethods = types_1.tuple('get', 'put', 'post', 'delete', 'options', 'head', 'patch');
