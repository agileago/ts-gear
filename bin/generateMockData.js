"use strict";
exports.__esModule = true;
// copy from swagger-ui and modified to ts
var lodash_1 = require("lodash");
// Deeply strips a specific key from an object.
//
// `predicate` can be used to discriminate the stripping further,
// by preserving the key's place in the object based on its value.
function deeplyStripKey(input, keyToStrip, predicate) {
    if (typeof input !== 'object' || Array.isArray(input) || input === null || !keyToStrip) {
        return input;
    }
    var obj = Object.assign({}, input);
    Object.keys(obj).forEach(function (k) {
        if (k === keyToStrip && predicate(obj[k], k)) {
            delete obj[k];
            return;
        }
        obj[k] = deeplyStripKey(obj[k], keyToStrip, predicate);
    });
    return obj;
}
exports.deeplyStripKey = deeplyStripKey;
function objectify(thing) {
    if (!lodash_1.isObject(thing))
        return {};
    return thing;
}
/* eslint-disable @typescript-eslint/camelcase */
var primitives = {
    string: function () { return 'string'; },
    string_email: function () { return 'user@example.com'; },
    'string_date-time': function () { return new Date('2019-09-03').toISOString(); },
    string_date: function () { return new Date('2019-09-03').toISOString().substring(0, 10); },
    string_uuid: function () { return '3fa85f64-5717-4562-b3fc-2c963f66afa6'; },
    string_hostname: function () { return 'example.com'; },
    string_ipv4: function () { return '198.51.100.42'; },
    string_ipv6: function () { return '2001:0db8:5b96:0000:0000:426f:8e17:642a'; },
    number: function () { return 0; },
    number_float: function () { return 0.0; },
    integer: function () { return 0; },
    boolean: function (schema) { return (typeof schema["default"] === 'boolean' ? schema["default"] : true); }
};
var primitive = function (schema) {
    schema = objectify(schema);
    var type = schema.type, format = schema.format;
    var key = type + "_" + format;
    var fn = primitives[key] || primitives[type];
    if (lodash_1.isFunction(fn))
        return fn(schema);
    throw new Error('Unknown Type: ' + schema.type);
};
/** schema的$ref会有互相包含的情况，需要检查循环引用 */
var schemaSet = new Set();
exports.sampleFromSchema = function (schema, definitions) {
    if (schemaSet.has(schema)) {
        return;
    }
    schemaSet.add(schema);
    var type = objectify(schema).type;
    var _a = objectify(schema), example = _a.example, properties = _a.properties, additionalProperties = _a.additionalProperties, items = _a.items, $ref = _a.$ref, schemaSchema = _a.schema;
    if (example !== undefined) {
        return deeplyStripKey(example, '$$ref', function (val) {
            // do a couple of quick sanity tests to ensure the value
            // looks like a $$ref that swagger-client generates.
            return typeof val === 'string' && val.indexOf('#') > -1;
        });
    }
    if (!type) {
        if (properties) {
            type = 'object';
        }
        else if (items) {
            type = 'array';
        }
        else if (definitions && $ref) {
            var definition = definitions[$ref];
            if (definition) {
                return exports.sampleFromSchema(definition, definitions);
            }
            return;
        }
        else if (schemaSchema) {
            return exports.sampleFromSchema(schemaSchema, definitions);
        }
        else {
            return;
        }
    }
    if (type === 'object') {
        var props = objectify(properties);
        var obj = {};
        for (var name_1 in props) {
            if (props[name_1] && props[name_1].deprecated) {
                continue;
            }
            obj[name_1] = exports.sampleFromSchema(props[name_1], definitions);
        }
        if (additionalProperties === true) {
            obj.additionalProp1 = {};
        }
        else if (additionalProperties) {
            var additionalProps = objectify(additionalProperties);
            var additionalPropVal = exports.sampleFromSchema(additionalProps, definitions);
            for (var i = 1; i < 4; i++) {
                obj['additionalProp' + i] = additionalPropVal;
            }
        }
        return obj;
    }
    if (type === 'array') {
        if (Array.isArray(items.anyOf)) {
            return items.anyOf.map(function (i) { return exports.sampleFromSchema(i, definitions); });
        }
        if (Array.isArray(items.oneOf)) {
            return items.oneOf.map(function (i) { return exports.sampleFromSchema(i, definitions); });
        }
        return [exports.sampleFromSchema(items, definitions)];
    }
    if (schema['enum']) {
        if (schema['default'])
            return schema['default'];
        return lodash_1.castArray(schema['enum'])[0];
    }
    if (type === 'file') {
        return;
    }
    return primitive(schema);
};
exports.generateMockData = lodash_1.memoize(exports.sampleFromSchema);
