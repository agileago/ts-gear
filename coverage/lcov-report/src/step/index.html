
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/step</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/step</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.24% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>195/290</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.17% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>59/120</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.07% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>191/285</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="assembleSchemaToGlobal.ts"><a href="assembleSchemaToGlobal.ts.html">assembleSchemaToGlobal.ts</a></td>
	<td data-value="96.55" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.55" class="pct high">96.55%</td>
	<td data-value="29" class="abs high">28/29</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="96.43" class="pct high">96.43%</td>
	<td data-value="28" class="abs high">27/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="cleanRefAndDefinitionName.ts"><a href="cleanRefAndDefinitionName.ts.html">cleanRefAndDefinitionName.ts</a></td>
	<td data-value="78.57" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	</tr>

<tr>
	<td class="file high" data-value="colletRefsInRequestAndPatchDefinition.ts"><a href="colletRefsInRequestAndPatchDefinition.ts.html">colletRefsInRequestAndPatchDefinition.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="fetchSwagger.ts"><a href="fetchSwagger.ts.html">fetchSwagger.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file low" data-value="generateDefinitionContent.ts"><a href="generateDefinitionContent.ts.html">generateDefinitionContent.ts</a></td>
	<td data-value="19.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	</tr>

<tr>
	<td class="file low" data-value="importAllDefinition.ts"><a href="importAllDefinition.ts.html">importAllDefinition.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="parseGenericType.ts"><a href="parseGenericType.ts.html">parseGenericType.ts</a></td>
	<td data-value="98.63" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 98%"></div><div class="cover-empty" style="width: 2%"></div></div>
	</td>
	<td data-value="98.63" class="pct high">98.63%</td>
	<td data-value="73" class="abs high">72/73</td>
	<td data-value="86.84" class="pct high">86.84%</td>
	<td data-value="38" class="abs high">33/38</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="98.57" class="pct high">98.57%</td>
	<td data-value="70" class="abs high">69/70</td>
	</tr>

<tr>
	<td class="file medium" data-value="prepareProjectDirectory.ts"><a href="prepareProjectDirectory.ts.html">prepareProjectDirectory.ts</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file low" data-value="translateSchema.ts"><a href="translateSchema.ts.html">translateSchema.ts</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="45" class="abs low">9/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="20.45" class="pct low">20.45%</td>
	<td data-value="44" class="abs low">9/44</td>
	</tr>

<tr>
	<td class="file low" data-value="writeProject.ts"><a href="writeProject.ts.html">writeProject.ts</a></td>
	<td data-value="35.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Apr 27 2020 18:43:36 GMT+0800 (GMT+08:00)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    