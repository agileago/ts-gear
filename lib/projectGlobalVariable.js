"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const clearObject_1 = require("./tool/clearObject");
const projectGlobal = {};
exports.getGlobal = (project) => {
    if (!projectGlobal[project.name]) {
        projectGlobal[project.name] = {
            definitionMap: {},
            requestMap: {},
            requestRefSet: new Set(),
        };
    }
    return projectGlobal[project.name];
};
exports.restore = (project) => {
    const g = projectGlobal[project.name];
    clearObject_1.clearObject(g.definitionMap);
    clearObject_1.clearObject(g.requestMap);
    g.requestRefSet.clear();
};
exports.httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
