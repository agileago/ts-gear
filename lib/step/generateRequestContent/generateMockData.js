"use strict";
/** copy from swagger-ui https://github.com/swagger-api/swagger-ui/blob/master/src/core/plugins/samples/fn.js
 * modify js to ts */
Object.defineProperty(exports, "__esModule", { value: true });
const lodash_1 = require("lodash");
// Deeply strips a specific key from an object.
//
// `predicate` can be used to discriminate the stripping further,
// by preserving the key's place in the object based on its value.
function deeplyStripKey(input, keyToStrip, predicate) {
    if (typeof input !== 'object' || Array.isArray(input) || input === null || !keyToStrip) {
        return input;
    }
    const obj = { ...input };
    Object.keys(obj).forEach(k => {
        if (k === keyToStrip && predicate(obj[k], k)) {
            delete obj[k];
            return;
        }
        obj[k] = deeplyStripKey(obj[k], keyToStrip, predicate);
    });
    return obj;
}
exports.deeplyStripKey = deeplyStripKey;
function objectify(thing) {
    if (!lodash_1.isObject(thing))
        return {};
    return thing;
}
/* eslint-disable @typescript-eslint/camelcase */
const primitives = {
    string: () => 'string',
    string_email: () => 'user@example.com',
    'string_date-time': () => new Date('2019-09-03').toISOString(),
    string_date: () => new Date('2019-09-03').toISOString().substring(0, 10),
    string_uuid: () => '3fa85f64-5717-4562-b3fc-2c963f66afa6',
    string_hostname: () => 'example.com',
    string_ipv4: () => '198.51.100.42',
    string_ipv6: () => '2001:0db8:5b96:0000:0000:426f:8e17:642a',
    number: () => 0,
    number_float: () => 0.0,
    integer: () => 0,
    boolean: (schema) => (typeof schema.default === 'boolean' ? schema.default : true),
};
const primitive = (schema) => {
    schema = objectify(schema);
    const { type, format } = schema;
    const key = `${type}_${format}`;
    const fn = primitives[key] || primitives[type];
    if (lodash_1.isFunction(fn))
        return fn(schema);
    throw new Error(`Unknown Type: ${schema.type}`);
};
/**
 * prevent schema circle reference
 * */
const schemaSet = new Set();
exports.sampleFromSchema = (schema, definitions) => {
    if (schemaSet.has(schema)) {
        return '';
    }
    schemaSet.add(schema);
    let { type } = objectify(schema);
    const { example, properties, additionalProperties, items, $ref, schema: schemaSchema } = objectify(schema);
    if (example !== undefined) {
        const r = deeplyStripKey(example, '$$ref', (val) => {
            // do a couple of quick sanity tests to ensure the value
            // looks like a $$ref that swagger-client generates.
            return typeof val === 'string' && val.indexOf('#') > -1;
        });
        return r;
    }
    if (!type) {
        if (properties) {
            type = 'object';
        }
        else if (items) {
            type = 'array';
        }
        else if ($ref) {
            const definitionSchema = definitions[$ref] && definitions[$ref].schema;
            if (definitionSchema) {
                return exports.sampleFromSchema(definitionSchema, definitions);
            }
            return '';
        }
        else if (schemaSchema) {
            return exports.sampleFromSchema(schemaSchema, definitions);
        }
        else {
            return '';
        }
    }
    if (type === 'object') {
        const props = objectify(properties);
        const obj = {};
        Object.getOwnPropertyNames(props).forEach(name => {
            if (!(props[name] && props[name].deprecated)) {
                obj[name] = exports.sampleFromSchema(props[name], definitions);
            }
        });
        if (additionalProperties === true) {
            obj.additionalProp1 = {};
        }
        else if (additionalProperties) {
            const additionalProps = objectify(additionalProperties);
            const additionalPropVal = exports.sampleFromSchema(additionalProps, definitions);
            for (let i = 1; i < 4; i += 1) {
                obj[`additionalProp${i}`] = additionalPropVal;
            }
        }
        return obj;
    }
    if (type === 'array') {
        if (Array.isArray(items.anyOf)) {
            return items.anyOf.map((i) => exports.sampleFromSchema(i, definitions));
        }
        if (Array.isArray(items.oneOf)) {
            return items.oneOf.map((i) => exports.sampleFromSchema(i, definitions));
        }
        return [exports.sampleFromSchema(items, definitions)];
    }
    if (schema.enum) {
        if (schema.default) {
            return schema.default;
        }
        return lodash_1.castArray(schema.enum)[0];
    }
    if (type === 'file') {
        return '';
    }
    return primitive(schema);
};
exports.generateMockData = lodash_1.memoize((request, definitionMap) => {
    schemaSet.clear();
    if (request.responses) {
        const schema = lodash_1.find(request.responses, (v, k) => k === 'default' || k.startsWith('2'));
        if (schema) {
            return exports.sampleFromSchema(schema, definitionMap);
        }
    }
    return '';
});
