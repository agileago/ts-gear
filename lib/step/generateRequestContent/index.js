"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateRequestContent = void 0;
const os_1 = require("os");
const source_1 = require("../../source");
const transformSwaggerPathToRouterPath_1 = require("../../tool/transformSwaggerPathToRouterPath");
const projectGlobalVariable_1 = require("../../projectGlobalVariable");
const assembleDoc_1 = require("../../tool/assembleDoc");
const constant_1 = require("../../constant");
const generateMockData_1 = require("./generateMockData");
const generateResponseType_1 = require("./generateResponseType");
const generateParameterType_1 = require("./generateParameterType");
const join = require("url-join");
/** from swagger spec paths assemble request functions */
exports.generateRequestContent = (spec, project) => {
    const { pathMatcher, withBasePath, withHost } = project;
    const { requestMap, definitionMap, enumMap } = projectGlobalVariable_1.getGlobal(project);
    const shouldMockResponseStatement = project.shouldMockResponseStatement || constant_1.defaultShouldMockResponseStatement;
    const resultContent = [];
    Object.getOwnPropertyNames(requestMap).forEach((requestFunctionName) => {
        const requestTypeScriptContent = [];
        const request = requestMap[requestFunctionName];
        const { httpMethod } = request;
        if (pathMatcher) {
            if (typeof pathMatcher === 'function') {
                if (!pathMatcher(request.pathName, httpMethod)) {
                    return;
                }
            }
            else if (!pathMatcher.test(request.pathName)) {
                return;
            }
        }
        let parameterTypeName = '';
        let parameterRequired = false;
        if (request.parameters && request.parameters.length > 0) {
            const parameterType = generateParameterType_1.generateParameterType(requestFunctionName, request.parameters);
            parameterTypeName = parameterType.parameterTypeName;
            parameterRequired = parameterType.parameterRequired;
            requestTypeScriptContent.push(parameterType.parameterTypeContent);
        }
        const responseType = generateResponseType_1.generateResponseType(requestFunctionName, request.responses);
        requestTypeScriptContent.push(responseType.responseTypeContent);
        requestTypeScriptContent.push(responseType.successTypeContent);
        const urlPath = join(spec.basePath || '', transformSwaggerPathToRouterPath_1.transformSwaggerPathToRouterPath(String(request.pathName)));
        const source = source_1.sow();
        const requesterStatment = `return requester('${urlPath}', {${withHost ? `, host: '${spec.host}'` : ''}${withBasePath ? `, basePath: '${spec.basePath}'` : ''}method: '${httpMethod}'${parameterTypeName ? ', ...option' : ''}}) as Promise<any>`;
        const functionStatment = `if (${shouldMockResponseStatement}) {
      return Promise.resolve(${requestFunctionName}.mockData as any)
    }
    ${requesterStatment}`;
        const functionData = {
            name: requestFunctionName,
            isExported: true,
            returnType: `Promise<${responseType.successTypeName}>`,
            statements: functionStatment,
            docs: assembleDoc_1.assembleDoc(request.schema),
        };
        functionData.parameters = [];
        if (parameterTypeName) {
            functionData.parameters.push({
                hasQuestionToken: !parameterRequired,
                name: 'option',
                type: parameterTypeName,
            });
        }
        source.addFunction(functionData);
        const mockStatment = `${requestFunctionName}.mockData = (${JSON.stringify(generateMockData_1.generateMockData(request, definitionMap, enumMap))} as any)`;
        source.addStatements(`
if (${shouldMockResponseStatement}) {
  ${mockStatment}
}
${requestFunctionName}.method = '${httpMethod}'
`);
        requestTypeScriptContent.push(source_1.harvest(source));
        /** store typescript content to requestMap */
        request.typescriptContent = requestTypeScriptContent.join(os_1.EOL);
        resultContent.push(request.typescriptContent);
    });
    /** return value only for test and debug */
    return resultContent.join(os_1.EOL);
};
