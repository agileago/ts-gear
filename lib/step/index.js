"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./getUserConfig"));
__export(require("./fetchSwagger"));
__export(require("./prepareProjectDirectory"));
__export(require("./translateSchema"));
__export(require("./assembleSchemaToGlobal"));
__export(require("./cleanRefAndDefinitionName"));
__export(require("./parseGenericType"));
__export(require("./colletRefsInRequestAndPatchDefinition"));
__export(require("./generateDefinitionContent"));
__export(require("./generateRequestContent"));
__export(require("./writeProject"));
