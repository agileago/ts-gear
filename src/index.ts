export { requester as fetchRequester } from './requester/fetch'
export { requester as axiosRequester } from './requester/axios'
export { IProject, PropertyType, IRequestParameter, Requester } from './interface'
