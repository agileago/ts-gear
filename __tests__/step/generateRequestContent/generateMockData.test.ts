import { cloneDeep } from 'lodash'
import { Spec } from 'swagger-schema-official'

import * as petSpec from 'example/fixture/pet.json'
import { generateMockData } from 'src/step/generateRequestContent/generateMockData'
import * as step from 'src/step'
import { IProject } from 'src/interface'
import { getGlobal, restore } from 'src/projectGlobalVariable'

describe('sample', () => {
  it('make sample data', () => {
    const project: IProject = {
      name: 'projectPont',
      source: 'fixture/pontFixture.json',
      dest: './service',
      // keepGeneric: true,
      translationEngine: 'baidu',
      requester: () => Promise.resolve({}),
    }

    const spec = cloneDeep(petSpec) as Spec
    step.cleanRefAndDefinitionName(spec, false)
    step.assembleSchemaToGlobal(spec, project)
    const { definitionMap, requestMap } = getGlobal(project)

    const result = generateMockData(requestMap.getPetFindByStatus, definitionMap)
    expect(result).toMatchSnapshot()

    restore(project)
  })
})
