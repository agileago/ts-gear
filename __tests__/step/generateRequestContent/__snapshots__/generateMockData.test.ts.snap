// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample make sample data 1`] = `
Array [
  Object {
    "category": Object {
      "id": 0,
      "name": "string",
      "pet": "",
    },
    "id": 0,
    "name": "doggie",
    "photoUrls": Array [
      "string",
    ],
    "status": "available",
    "tags": Array [
      Object {
        "id": 0,
        "name": "string",
      },
    ],
  },
]
`;
